resolver/src/p2p/mod.rs

```diff
@@ -0,0 +1,544 @@
+        context: &Context,
+        peer_seeds: &[u64],
+    ) -> (
+        Oracle<PublicKey>,
+        Vec<Ed25519>,
+        Vec<PublicKey>,
+        Vec<(Sender<PublicKey>, Receiver<PublicKey>)>,
+    ) {
+        let (network, mut oracle) = Network::new(
+            context.with_label("network"),
+            commonware_p2p::simulated::Config {
+                max_size: 1024 * 1024,
+            },
+        );
+        network.start();
+
+        let schemes: Vec<Ed25519> = peer_seeds
+            .iter()
+            .map(|seed| Ed25519::from_seed(*seed))
+            .collect();
+        let peers: Vec<PublicKey> = schemes.iter().map(|s| s.public_key()).collect();
+
+        let mut connections = Vec::new();
+        for peer in &peers {
+            let (sender, receiver) = oracle.register(peer.clone(), 0).await.unwrap();
+            connections.push((sender, receiver));
+        }
+
+        (oracle, schemes, peers, connections)
+    }
+
+    async fn add_link(
+        oracle: &mut Oracle<PublicKey>,
+        link: Link,
+        peers: &[PublicKey],
+        from: usize,
+        to: usize,
+    ) {
+        oracle
+            .add_link(peers[from].clone(), peers[to].clone(), link.clone())
+            .await
+            .unwrap();
+        oracle
+            .add_link(peers[to].clone(), peers[from].clone(), link)
+            .await
+            .unwrap();
+    }
+
+    #[allow(clippy::type_complexity)]
+    fn setup_consumer() -> (
+        Consumer<Key, Bytes, ()>,
+        mpsc::Receiver<Event<Key, Bytes, ()>>,
+    ) {
+        let (sender, receiver) = mpsc::channel(MAILBOX_SIZE);
+        let consumer = Consumer::new(sender);
+        (consumer, receiver)
+    }
+
+    async fn setup_and_spawn_actor(
+        context: &Context,
+        director: &Director<PublicKey>,
+        scheme: Ed25519,
+        connection: (Sender<PublicKey>, Receiver<PublicKey>),
+        consumer: Consumer<Key, Bytes, ()>,
+        producer: Producer<Key, Bytes>,
+    ) -> peer::Mailbox<Key> {
+        let public_key = scheme.public_key();
+        let (actor, mailbox) = peer::Actor::new(
+            context.with_label(&format!("actor_{}", scheme.public_key())),
+            peer::Config {
+                crypto: scheme.clone(),
+                director: director.clone(),
+                consumer,
+                producer,
+                mailbox_size: MAILBOX_SIZE,
+                requester_config: commonware_p2p::utils::requester::Config {
+                    crypto: scheme,
+                    rate_limit: governor::Quota::per_second(
+                        std::num::NonZeroU32::new(RATE_LIMIT).unwrap(),
+                    ),
+                    initial: INITIAL_DURATION,
+                    timeout: TIMEOUT,
+                },
+                fetch_retry_timeout: FETCH_RETRY_TIMEOUT,
+            },
+        )
+        .await;
+
+        let (sender, receiver) = connection;
+        context
```

Comment:
You should just call `actor.start(connection)` here? I'm not sure why it would need to be wrapped in another spawn?